pyro/distributions/half_cauchy.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import absolute_import, division, print_function
+
+import math
+
+from torch.distributions import constraints
+from torch.distributions.transforms import AbsTransform, AffineTransform
+from torch.distributions.utils import broadcast_all
+
+from pyro.distributions.torch import Cauchy, TransformedDistribution
+
+
+class HalfCauchy(TransformedDistribution):
+    r"""
+    Half-Cauchy distribution.
+
+    This is a continuous distribution with lower-bounded domain (`x > loc`).
+    See also the :class:`~pyro.distributions.torch.Cauchy` distribution.
+
+    :param torch.Tensor loc: lower bound of the distribution.
+    :param torch.Tensor scale: half width at half maximum.
+    """
+    params = Cauchy.params
+
+    def __init__(self, loc, scale):
+        loc, scale = broadcast_all(loc, scale)
+        base_dist = Cauchy(0, scale)
+        transforms = [AbsTransform(), AffineTransform(loc, 1)]
+        super(HalfCauchy, self).__init__(base_dist, transforms)
+
+    @property
+    def loc(self):
+        return self.transforms[1].loc
+
+    @property
+    def scale(self):
+        return self.base_dist.scale
+
+    @constraints.dependent_property
+    def support(self):
+        return constraints.greater_than(self.loc)
```

Comment:
I just wonder if we set support `greater_than(0)` then is `0` still a valid value for this distribution?

Reply:
yeah it technically should be "greater than or equal to" though im not clear it makes a difference for continuous distributions